### PR TITLE
Update to Rails 8.0.1

### PR DESCRIPTION
**Pull request by DavidBiddle**
### What problem does this pull request solve?

Trello card: 

Updated by:

- taking the initial Dependabot update commit
- running `bin/rails app:update` and reviewing the changes against RailsDiff and the commit history to avoid overwriting our deliberate changes
- fixing an issue caused by an APi deprecation

See individual commit messages for details.

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

**Review by stephencdaly — APPROVED** [review]
Changes all look sensible enough to me and can't spot any issues when running locally. Nice work!

File: config/environments/production.rb [hunk 1]
```diff
@@ -63,36 +48,50 @@
...
   config.i18n.fallbacks = true
 
-  # Don't log any deprecations.
-  config.active_support.report_deprecations = false
-
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 
+  # Only use :id for inspections in production.
+  config.active_record.attributes_for_inspect = [:id]
```
Comment (chao-xian) [inline config/environments/production.rb:87]: Where did this change come from? It's part of https://railsdiff.org/7.2.2/8.0.1

Comment (DavidBiddle) [reply config/environments/production.rb:87]: Good question - it definitely came from rails app:update but looks like it was introduced in the v7.2.1 -> v7.2.2 bump: https://railsdiff.org/7.2.1/7.2.2

Comment (DavidBiddle) [reply config/environments/production.rb:87]: We do use `inspect` in one of our rake tasks so I think I need to remove this line or set it to `:all`

Comment (chao-xian) [reply config/environments/production.rb:87]: so we must have taken it out in the past?

Comment (DavidBiddle) [reply config/environments/production.rb:87]: Or we accepted the 7.2.2 dependabot bump without running the script

Comment (DavidBiddle) [reply config/environments/production.rb:87]: PR for removing it here anyway: https://github.com/alphagov/forms-admin/pull/1673

**Pull request merged**